backend/tests/integration-async/migrator/test_migrator_tasks.py

```diff
@@ -593,3 +590,62 @@ def test_migrate_recids_from_mirror_all_only_with_literature_author_and_invalid(
 
     with pytest.raises(PIDDoesNotExistError):
         InspireRecord.get_record_by_pid_value(invalid_control_number, "lit")
+
+
+def test_process_references_in_records_reindexes_experiments_when_linked_experiments_change(
+    app, celery_app_with_context, celery_session_worker
+):
+    experiment_data = faker.record("exp", with_control_number=True)
+    experiment = InspireRecord.create(experiment_data)
+    db.session.commit()
+
+    experiment_control_number = experiment["control_number"]
+    exp_ref = f"http://localhost:8000/api/experiments/{experiment_control_number}"
+
+    data = faker.record("lit", with_control_number=True)
+
+    data["accelerator_experiments"] = [
+        {"legacy_name": "LIGO", "record": {"$ref": exp_ref}}
+    ]
+
+    record = InspireRecord.create(data, disable_relations_update=True)
+    db.session.commit()
+
+    task = update_relations.delay([record.id])
```

Comment:
the test says it is testing the ```process_references_in_records``` not the ```update_relations```. You should also disconnect the signal to ES because otherwise the test is useless. You are asserting that the record is indexed but it would have been indexed from the beginning without calling the reindex task.